docs/DeveloperGuide.md

```diff
@@ -2,37 +2,731 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+LongAh uses the following libraries:
 
-## Design & implementation
+1. [XChart](https://knowm.org/open-source/xchart/) - Used for generating charts to visualize data.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+LongAh uses the following tools for development:
 
+1. [JUnit 5](https://junit.org/junit5/) - Used for testing.
+2. [Gradle](https://gradle.org/) - Used for build automation.
+
+## Table of Contents
+- [Developer Guide](#developer-guide)
+  - [Acknowledgements](#acknowledgements)
+  - [Table of Contents](#table-of-contents)
+  - [Design \& Implementation](#design--implementation)
+    - [UI and I/O](#ui-and-io)
+    - [Commands](#commands)
+    - [Storage](#storage)
+    - [Member and MemberList](#member-and-memberlist)
+    - [Transaction and TransactionList](#transaction-and-transactionlist)
+    - [PIN](#pin)
+    - [Chart](#chart)
+    - [Exceptions and Logging](#exceptions-and-logging)
+  - [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+  - [User Stories](#user-stories)
+  - [Non-Functional Requirements](#non-functional-requirements)
+  - [Glossary](#glossary)
+  - [Instructions for manual testing](#instructions-for-manual-testing)
+  - [Instructions for JUnit Testing](#instructions-for-junit-testing)
+  - [Instructions for text-ui-testing](#instructions-for-text-ui-testing)
+  - [Future Enhancements](#future-enhancements)
+
+## Design & Implementation
+
+```plantuml
+@startuml
+left to right direction
+
+abstract class longah.commands.Command {
++ {abstract} execute(Group)
+}
+
+class longah.node.Transaction {
```

Comment:
You could try and save the .puml as an image, and embed that image into the .md file.